File: Graph/637. Average of Levels in Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        vector<double> ans;
        vector<TreeNode*> tmp;
        vector<TreeNode*> cur;
        cur.insert(cur.begin(), root);
        double s;

        while (!cur.empty()){
            s = 0;
            for (auto i : cur){
                s += i->val;
                if (i->left)
                    tmp.insert(tmp.end(), i->left);
                if (i->right)
                    tmp.insert(tmp.end(), i->right);
            }
            ans.insert(ans.end(), s / cur.size());
            cur = tmp;
            tmp.clear();
        }
        
        return ans;
    }
};

File: Graph/637. Average of Levels in Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        ans = []
        cur = [root]
        tmp = []
        while cur:
            s = 0
            for i in cur:
                s += i.val
                if i.left:
                    tmp.append(i.left)
                if i.right:
                    tmp.append(i.right)
            ans.append(s / len(cur))
            cur = tmp
            tmp = []
        return ans